cv_next/app/feed/components/CVItemRSC.tsx

```diff
@@ -1,18 +1,35 @@
-import Image from "next/image"
-import Link from "next/link"
-import generateImageUrl from "@/helpers/imageURLHelper"
-import Categories from "@/types/models/categories"
-import { ClientCvModel } from "@/types/models/cv"
+"use client";
+import Image from "next/image";
+import Link from "next/link";
+import generateImageUrl from "@/helpers/imageURLHelper";
+import Categories from "@/types/models/categories";
+import { useEffect, useState, cache } from "react";
+import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";
+
+const getBlur = cache(async (imageURL: string) => {
+  return await getBlurredCv(imageURL);
+});
 interface CVCardProps {
-  cv: ClientCvModel
+  cv: CvModel;
 }
 
 export default function CVItemRSC({ cv }: CVCardProps) {
+  const [base64Data, setBase64Data] = useState(
```

Comment:
By adding this you've forced it to be client component :)
you can simply do something like `getBlur(imageUrl) ?? '...';` instead of introducing state.
By the way, because this component is a server component, the browser should already receive it with the data loaded - that's the point of loading the data on the server. It's not the browser sending the request, it's the server sending it, before sending the component (html/js code) to the browser. So you should be fine with:
```Suggestion
const base64Data = await getBlur(imageUrl);
```

Just make sure that `getBlur` returns string even on failure.

Reply:
Tried doing it the way you mentioned(removed all references to useState/Effect, added await to getBlur and async near the export of the export, etc) but it said that async/await is only supported in server components and every time new items loaded, the entire thing disappeared. Is that the way you meant?
When I tried adding use server it crashed because it couldn't handle the Props.

Reply:
Show me a) how the file looks now and b) how the component that calls it looks

Reply:
The RSC, with use server at the top
```ts
import { cache } from "react";
import { getBlurredCv } from "@/app/actions/cvs/getBlurCv";

const getBlur = cache(async (imageURL: string) => {
  return await getBlurredCv(imageURL);
});

interface CVCardProps {
  cv: CvModel;
}

export default async function CVItemRSC({ cv }: CVCardProps) {
  const categoryLink = `/feed?category=${Categories.category[
    cv.category_id
  ].toLowerCase()}`;
  const imageUrl = generateImageUrl(cv.document_link);
  const formattedDate = new Date(cv.created_at).toLocaleDateString("en-US");
  const base64Data =
    (await getBlur(imageUrl)) ??
    "data:iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7p5fDwAFlAI2LB7hbAAAAABJRU5ErkJggg==";

  return (...)
```
The ... at the return is just the image with the data the also needs to be with the image(category, uid, etc)

What's calling in is the feed that is pretty long but the relevant part is
```ts
cvs.map((cv) => (
              <CVItem key={cv.id} cv={cv}>
                <CVItemRSC cv={cv} />
              </CVItem>
            )
```

I can add the files if you'd like

Reply:
The reason I wanted to see the file you import it in, is because client and server interleaving is a bit complicated. Make sure you follow this pattern: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props.
I recommend you read the whole page and make sure you understand it.

Reply:
Read it several times, couldn't understand how to send the props to CVItemRSC from the parent(or skip it and pass only to the children)